shimmingtoolbox/unwrap/prelude.py

```diff
@@ -0,0 +1,82 @@
+    the complex image provided. 2D unwrapping can be turned off. The output path can be specified. THe temporary niis
+    can optionally be saved.
+
+    Args:
+        complex_array (numpy.ndarray): 3D complex values numpy array to perform phase unwrapping
+        affine (numpy.ndarray): 2D array (4x4) containing the transformation coefficients. Can be acquired by :
+            nii = nib.load("nii_path")
+            affine = nii.affine
+        mask (numpy.ndarray): numpy array of booleans with shape of `complex_array` to mask during phase unwrapping
+        path_2_unwrapped_phase (string): relative or absolute path to output the nii unwrapped phase
+        is_unwrapping_in_2d (bool): prelude parameter to unwrap un 2d
+        is_saving_nii (bool): specify whether `complex_array`, `affine`, `mask` and `unwrapped_phase` nii files will be
+        saved
+
+    Returns:
+        numpy.ndarray: 3D array with the shape of `complex_array` of the unwrapped phase output from prelude
+    """
+    # Get absolute path
+    abs_path = os.path.abspath(path_2_unwrapped_phase)
+    data_save_directory = os.path.dirname(abs_path)
+
+    # Save complex image TODO: Save voxelSize when saving nifti (Maybe affine does it) (if it does, might be easier
+    #  to just pass entire nibabel object)
+    phase_nii = nib.Nifti1Image(np.angle(complex_array), affine)
+    mag_nii = nib.Nifti1Image(np.abs(complex_array), affine)
+    nib.save(phase_nii, os.path.join(data_save_directory, 'rawPhase.nii'))
+    nib.save(mag_nii, os.path.join(data_save_directory, 'mag.nii'))
+
+    # Fill options
+    options = ' '
+    if is_unwrapping_in_2d:
+        options = '-s '
+
+    # Add mask data and options if there is a mask provided
+    if not np.any(mask == -1):
+        # TODO: Make sure values are either 1 or 0
+        if mask.shape != complex_array.shape:
+            raise Exception('Mask must be the same shape as the array')
+        mask_nii = nib.Nifti1Image(mask, affine)
+
+        options += '-m '
+        options += os.path.join(data_save_directory, 'mask.nii')
+        nib.save(mask_nii, os.path.join(data_save_directory, 'mask.nii'))
+
+    # Unwrap
+    unwrap_command = 'prelude -p {} -a {} -o {} {}'.format(os.path.join(data_save_directory, 'rawPhase'),
+                                                           os.path.join(data_save_directory, 'mag'),
+                                                           path_2_unwrapped_phase, options)
+    subprocess.run(unwrap_command, shell=True, check=True)
+
+    # Uncompress
+    subprocess.run(['gunzip', path_2_unwrapped_phase + '.gz', '-df'], check=True)
+    unwrapped_phase = nib.load(path_2_unwrapped_phase)
+    unwrapped_phase = unwrapped_phase.get_fdata()
+
+    # Delete temporary files according to options
+    if not is_saving_nii:
+        os.remove(os.path.join(data_save_directory, 'mag.nii'))
```

Comment:
When removing files, you have to be carefull. If the file is used by another process, in Windows, it will crash (you'd have to manage this error), but in Linux, it will delete it without freeing the memory allocated until the file is closed (wich could result in memory loss if not well coordinated). This is why it is important to think about the memory allocated during your program. A simple variable, will be deleted at the end by the garbage collector. However, dynamic entities like pointers or file descriptor can leak memory. A way to remove these ambiguities is to check if the file is opened in another process and refuse to delete if it is (with a warning) or wait to close it.
psutil can help you to manage these sort of mistakes. Something like this for one file (I'm not 100% sure how it is, but should be something like this):
```suggestion
        for proc in psutil.process_iter():
             flist = proc.open_files()
             if os.path.join(data_save_directory, 'mag.nii') not in flist:
                     os.remove(os.path.join(data_save_directory, 'mag.nii'))
```

It begins to be a deep understanding of programming language and is a good practice for packages and programs to not leak memory. However, it is your desicion to choose if you want to implement it or not.